### PR TITLE
Add HTMLSerializer and Elements types to prismic-dom

### PR DESCRIPTION
**Pull request by kavehsajjadi**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
  - [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
  - [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/prismicio/prismic-dom/blob/fe798dd6c8a0d4a60fbc310e991419d7573380fc/src/richtext.js#L85>
  - [x] Increase the version number in the header if appropriate.
  - [ ] If you are making substantial changes, consider adding a tslint.json containing { "extends": "dtslint/dt.json" }.

### CODE REVIEW

**Comment by kavehsajjadi** [comment]
This definition package is for `prismic-dom`.
`prismic-dom` (which is not typed) internally uses `prismic-richtext` (which is typed). 
All the types provided in this package are available in `prismic-richtext`, can this `@typed` package import and expose `prismic-richtext` types?
@nickw444 I'm not sure who to tag in this comment, would you mind tagging relevant maintainers?

**Comment by kavehsajjadi** [comment]
@nickw444 I did my best trying to find a way to use the actual prismic-richtext types but as it turns out either I suck at TS (mostly true) and/or it can't be done too well because of how they wrote their types (maybe true?)

If you're okay with what I've done so far pls merge - it seems to work

File: types/prismic-dom/index.d.ts [hunk 1]
```diff
@@ -18,3 +32,25 @@ export const Link: Link;
 
 declare const _default: { RichText: RichText, Link: Link };
 export default _default;
+
+export type Elements = {
```
Comment (nickw444) [inline types/prismic-dom/index.d.ts]: Are you definitely sure you can't re-export this from the `richtext` library:

https://github.com/prismicio/prismic-richtext/blob/master/src/types.ts#L3-L23

I think you'll need to do a `typeof NODE_TYPES` to get the correct shape of `NODE_TYPES`, but I am not 100% sure.

**Review by nickw444 — CHANGES_REQUESTED** [review]

File: types/prismic-dom/index.d.ts (hunk 1 above)
Comment (kavehsajjadi) [reply types/prismic-dom/index.d.ts]:
```
...
../ramda/index.d.ts(2688,40): error TS2304: Cannot find name 'T'.
../ramda/index.d.ts(2688,46): error TS2304: Cannot find name 'U'.
../ramda/index.d.ts(2688,52): error TS2304: Cannot find name 'TResult'.
../ramda/index.d.ts(2688,60): error TS1005: ';' expected.
../ramda/index.d.ts(2688,84): error TS2304: Cannot find name 'T'.
../ramda/index.d.ts(2688,109): error TS2304: Cannot find name 'U'.
../ramda/index.d.ts(2688,116): error TS2304: Cannot find name 'TResult'.
../ramda/index.d.ts(2689,5): error TS1128: Declaration or statement expected.
../ramda/index.d.ts(2690,1): error TS1128: Declaration or statement expected.
```

**Pull request closed without merging**